eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala

```diff
@@ -258,25 +287,68 @@ case class PaymentRequest(
     tags: Vector[JsObject],
     signature: LnInvoiceSignature)
 
-sealed abstract class PaymentResult
-case class PaymentSucceeded(
-    amountMsat: MilliSatoshis,
-    paymentHash: Sha256Digest,
-    paymentPreimage: String,
-    route: JsArray)
-    extends PaymentResult
-
-case class PaymentFailed(paymentHash: Sha256Digest, failures: Vector[JsObject])
-    extends PaymentResult
-/*
-case class PaymentFailure(???) extends SendResult
-implicit val paymentFailureReads: Reads[PaymentFailure] = Json.reads[PaymentFailure]
-implicit val sendResultReads: Reads[SendResult] = Reads[SendResult] { json =>
-  json.validate[PaymentSucceeded] match {
-    case success: JsSuccess[PaymentSucceeded] => success
-    case err1: JsError => json.validate[PaymentFailure] match {
-      case failure: JsSuccess[PaymentFailure] => failure
-      case err2: JsError => JsError.merge(err1, err2)
-    }
+case class PaymentResult(
+  id: String,
+  paymentHash: Sha256Digest,
+  preimage: Option[PaymentPreimage],
+  amountMsat: MilliSatoshis,
+  createdAt: FiniteDuration,
+  completedAt: Option[FiniteDuration],
+  status: PaymentStatus)
+
+case class ReceivedPaymentResult(
+  paymentHash: Sha256Digest,
+  amountMsat: MilliSatoshis,
+  receivedAt: FiniteDuration)
+
+sealed trait PaymentStatus
+object PaymentStatus {
+  case object PENDING extends PaymentStatus
+  case object SUCCEEDED extends PaymentStatus
+  case object FAILED extends PaymentStatus
+
+  def apply(s: String): PaymentStatus = s match {
+    case "PENDING" => PENDING
+    case "SUCCEEDED" => SUCCEEDED
+    case "FAILED" => FAILED
+    case _ => throw new IllegalArgumentException("Invalid payment status code")
```

Comment:
do this do the error is helpful as to what string we didn't match if we miss a type

```scala
case err: String => throw new IllegalArgumentException("Unknown payment status=${err}")
```